modules/developer-cli-odo-creating-a-java-application-by-using-a-devfile-in-a-disconnected-cluster.adoc

```diff
@@ -0,0 +1,188 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/developer_cli_odo/
+
+[id="developer-cli-odo-creating-a-java-application-by-using-a-devfile-in-a-disconnected-cluster_{context}"]
+
+= Creating a Java application by using a devfile in a disconnected cluster
+
+.Prerequisites
+* You have created and logged into a disconnected cluster.
+* You have added `quay.io`, `registry.access.redhat.com`, `apache.org`, `quayio-production-s3.s3.amazonaws.com` URLs in your proxy.
+
+.Procedure
+
+. Define your Java application in a devfile:
++
+.Example of a devfile
+[source,yaml]
+----
+schemaVersion: 2.0.0
+metadata:
+  name: java-maven
+  version: 1.1.0
+starterProjects:
+  - name: springbootproject
+    git:
+      remotes:
+        origin: "https://github.com/odo-devfiles/springboot-ex.git"
+components:
+  - name: tools
+    container:
+      image: quay.io/eclipse/che-java11-maven:nightly
+      memoryLimit: 512Mi
+      mountSources: true
+      endpoints:
+        - name: 'http-8080'
+          targetPort: 8080
+      volumeMounts:
+        - name: m2
+          path: /home/user/.m2
+  - name: m2
+    volume: {}
+commands:
+  - id: mvn-package
+    exec:
+      component: tools
+      commandLine: "mvn -Dmaven.repo.local=/home/user/.m2/repository -Dhttp.proxyHost=<proxy-host> -Dhttp.proxyPort=<proxy-port> -Dhttps.proxyHost=<proxy-host> -Dhttps.proxyPort=<proxy-port> package"
+      group:
+        kind: build
+        isDefault: true
+  - id: run 
+    exec:
+      component: tools
+      commandLine: "java -jar target/*.jar"
+      group:
+        kind: run
+        isDefault: true
+  - id: debug
+    exec:
+      component: tools
+      commandLine: "java -Xdebug -Xrunjdwp:server=y,transport=dt_socket,address=${DEBUG_PORT},suspend=n -jar target/*.jar"
+      group:
+        kind: debug
+        isDefault: true
+----
+
+. Create a Java application:
++
+[source,terminal]
+----
+$ odo create --devfile <path-to-your-devfile> --starter
```

Comment:
I think it should be `odo create <cmp-name> --starter`, no?

Reply:
Even though users provide their own similar devfile (which has starter project) they need to pass component name, I think

Reply:
@prietyc123 thank you! I've updated the commands and also added the warnings about the external dependencies (hopefully, the support will allow these). PTAL and let me know if you're okay with the changes.

Reply:
@boczkowska Thanks for the changes 🙂 
LGTM